config/pipelines/tasks/meteor_buildah.yaml

```diff
@@ -0,0 +1,284 @@
+    container registry.
+
+    Buildah Task builds source into a container image using Project Atomic's
+    Buildah build tool.It uses Buildah's support for building from Dockerfiles,
+    using its buildah bud command.This command executes the directives in the
+    Dockerfile to assemble a container image, then pushes that image to a
+    container registry.
+  params:
+    - description: Reference of the image buildah will produce.
+      name: IMAGE
+      type: string
+    - default: >-
+        registry.redhat.io/rhel8/buildah@sha256:6a68ece207bc5fd8db2dd5cc2d0b53136236fb5178eb5b71eebe5d07a3c33d13
+      description: The location of the buildah builder image.
+      name: BUILDER_IMAGE
+      type: string
+    - default: vfs
+      description: Set buildah storage driver
+      name: STORAGE_DRIVER
+      type: string
+    - default: ./Dockerfile
+      description: Path to the Dockerfile to build.
+      name: DOCKERFILE
+      type: string
+    - default: .
+      description: Path to the directory to use as context.
+      name: CONTEXT
+      type: string
+    - default: "true"
+      description: >-
+        Verify the TLS on the registry endpoint (for push/pull to a non-TLS
+        registry)
+      name: TLSVERIFY
+      type: string
+    - default: oci
+      description: "The format of the built container, oci or docker"
+      name: FORMAT
+      type: string
+    - default: ""
+      description: Extra parameters passed for the build command when building images.
+      name: BUILD_EXTRA_ARGS
+      type: string
+    - default: ""
+      description: Extra parameters passed for the push command when pushing images.
+      name: PUSH_EXTRA_ARGS
+      type: string
+    - name: BASE_IMAGE
+
+  workspaces:
+  - name: data
+  - name: sslcertdir
+    optional: true
+    mountPath: /workspace/sslcertdir
+
+  results:
+    - name: IMAGE_DIGEST
+      description: Digest of the image just built.
+    - name: IMAGE_URL
+      description: URL  of the built Image.
+
+  steps:
+    - image: $(params.BUILDER_IMAGE)
+      name: build
+      workingDir: $(workspaces.data.path)
+      resources:
+        limits:
+          cpu: "2"
+          memory: "8Gi"
+        requests:
+          cpu: "1"
+          memory: "2Gi"
+      script: |
+        mkdir -p .tekton_metrics
+        touch .tekton_metrics/images
+        start=`date +%s`
+
+        [[ "$(workspaces.sslcertdir.bound)" == "true" ]] && CERT_DIR_FLAG="--cert-dir $(workspaces.sslcertdir.path)"
+        buildah ${CERT_DIR_FLAG}  --storage-driver=$(params.STORAGE_DRIVER) bud \
+          $(params.BUILD_EXTRA_ARGS) --format=$(params.FORMAT) \
+          --tls-verify=$(params.TLSVERIFY) --no-cache \
+          -f $(params.DOCKERFILE) -t $(params.IMAGE) $(params.CONTEXT)
+
+        end=`date +%s`
+
+        touch .tekton_metrics/image_build_success
+        echo "$((end-start))" > .tekton_metrics/image_build_duration
```

Comment:
It feels a bit wrong to mix metrics with the actual tasks. Can't we use the metrics from tekton (https://tekton.dev/docs/pipelines/metrics/) ?

Reply:
Another thing about buildah (more precisely what we run inside it): I think we probably don't want to run it with the same service account as the steps creating ImageStream (or accessing the Openshift API somehow). Since sdist can run arbitrary code when built (and build is happening as part of installation for sdist)

Reply:
yes, we do used tekton metrics
at the point of time of meteor, these metrics seemed to be something worth.
i will this and include the clustertask ref.